libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -107,6 +107,51 @@ string YulUtilFunctions::copyToMemoryFunction(bool _fromCalldata)
 	});
 }
 
+string YulUtilFunctions::copyLiteralToMemoryFunction(string const& _literal)
+{
+	string functionName = "copy_literal_to_memory_" + util::toHex(util::keccak256(_literal).asBytes());
+
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>() -> memPtr {
+				memPtr := <arrayAllocationFunction>(<size>)
+				<storeLiteralInMem>(add(memPtr, 32))
+			}
+			)")
+			("functionName", functionName)
+			("arrayAllocationFunction", allocateMemoryArrayFunction(*TypeProvider::array(DataLocation::Memory, true)))
+			("size", to_string(_literal.size()))
+			("storeLiteralInMem", storeLiteralInMemoryFunction(_literal))
+			.render();
+	});
+}
+
+string YulUtilFunctions::storeLiteralInMemoryFunction(string const& _literal)
+{
+	string functionName = "store_literal_to_memory_" + util::toHex(util::keccak256(_literal).asBytes());
```

Comment:
```suggestion
	string functionName = "store_literal_in_memory_" + util::toHex(util::keccak256(_literal).asBytes());
```

I'm not sure if we ensure that the yul helper function name matches the c++ one, but maybe it is a good idea to do so.